### PR TITLE
GOLD-277 : Required minimum for consensus should be STATIC not dynamic number

### PR DESCRIPTION
**Pull request by abdulazeem-tk4vr**
https://linear.app/shm/issue/GOLD-277/required-minimum-for-consensus-should-be-static-not-dynamic-number

### CODE REVIEW

**Comment by abdulazeem-tk4vr** [comment]
This is the final change that is needed imo. If the execution group length has a lower size, we throw an error else we use the larger value. Because if execution group is not smaller, it will then use the value it has which is larger than the minimumConsensusGroup size.
image
image

**Comment by abdulazeem-tk4vr** [comment]
> This is the final change that is needed imo. If the execution group length has a lower size, we throw an error else we use the larger value. Because if execution group is not smaller, it will then use the value it has which is larger than the minimumConsensusGroup size. image image

The team decided that we only need to make changes where the consensus decisions were being made. There were other places where we share the information after consensus and we concluded that it wouldn't be an issue

File: src/state-manager/TransactionQueue.ts [hunk 1]
```diff
@@ -2182,6 +2182,19 @@ class TransactionQueue {
           //set the nodes that are in the executionGroup.
           //This is needed so that consensus will expect less nodes to be voting
           const unRankedExecutionGroup = homeShardData.homeNodes[0].consensusNodeForOurNodeFull.slice()
+
+          // here we  reject the transaction if the executionGroup.length is lesser than minimumConsensusGroup size, so that we dont have to go ahead with the remaining processing and hereby maintaining the safe node count for consensus
+          if (txQueueEntry.executionGroup.length < this.config.sharding.minNodesPerConsensusGroup) {
```
Comment (aniketdivekar) [inline src/state-manager/TransactionQueue.ts:2211]: @abdulazeem-tk4vr is it possible that execution group length is greater than `minNodesPerConsensusGroup`? If so, is that fine in the premise of this ticket?

Comment (abdulazeem-tk4vr) [reply src/state-manager/TransactionQueue.ts:2211]: yes that is fine.

**Comment by aniketdivekar** [comment]
closing in favour of this [PR](https://github.com/shardeum/shardus-core/pull/339)

**Pull request closed without merging**